### ISSUE TITLE
Cannot use nfs filesystem or configure networks with photon

### ISSUE THREAD

**Issue by eatstoomuchjam** [mitchellh/vagrant#5618]
I'm just tried switching one of my vagrant setups over to photon since it was released yesterday.  I've run into two problems so far.  First, I was using NFS to share filesystems to the VM and received an error about the VM being unable to use NFS.  I logged into the VM to look around and it would appear that `mount.nfs*` are not present, though the kernel modules to support NFS are.  I'm not sure yet if this is an omission from VMware in Photon or an option not installed in the "vmware/photon" image.

```
bash-3.2$ vagrant up
Bringing machine 'default' up with 'vmware_fusion' provider...
==> default: Cloning VMware VM: 'vmware/photon'. This can take some time...
==> default: Checking if box 'vmware/photon' is up to date...
==> default: Verifying vmnet devices are healthy...
The synced folder type 'nfs' is reporting as unusable for
your current setup. Please verify you have all the proper
prerequisites for using this shared folder type and try again.
```

And second, even after I commented out just about everything in my Vagrantfile related to networking, attempting to bring it up gives an error about inability to execute `configure_networks`

```
    default: Are you sure you want to destroy the 'default' VM? [y/N] y
==> default: Stopping the VMware VM...
==> default: Deleting the VM...
==> default: Running cleanup tasks for 'shell' provisioner...
==> default: Running cleanup tasks for 'shell' provisioner...
==> default: Running cleanup tasks for 'shell' provisioner...
==> default: Running cleanup tasks for 'shell' provisioner...
==> default: Running cleanup tasks for 'shell' provisioner...
==> default: Running cleanup tasks for 'shell' provisioner...
Bringing machine 'default' up with 'vmware_fusion' provider...
==> default: Cloning VMware VM: 'vmware/photon'. This can take some time...
==> default: Checking if box 'vmware/photon' is up to date...
==> default: Verifying vmnet devices are healthy...
==> default: Preparing network adapters...
==> default: Fixed port collision for 22 => 2222. Now on port 2200.
==> default: Starting the VMware VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 192.168.159.174:22
    default: SSH username: vagrant
    default: SSH auth method: private key
    default:
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default:
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if its present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
==> default: Forwarding ports...
    default: -- 22 => 2200
==> default: Configuring network adapters within the VM...
Vagrant attempted to execute the capability 'configure_networks'
on the detect guest OS 'linux', but the guest doesn't
support that capability. This capability is required for your
configuration of Vagrant. Please either reconfigure Vagrant to
avoid this capability or fix the issue by creating the capability.
```

Vagrantfile looks like this networking stuff removed.  I'll try pulling the `vagrant-cachier` stuff to see if maybe that gets rid of the networking error message.

```
# -*- mode: ruby -*-
# vi: set ft=ruby :

# Vagrantfile API/syntax version. Don't touch unless you know what you're doing!
VAGRANTFILE_API_VERSION = "2"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.synced_folder "../", "/vagrant/"

  config.vm.box = "vmware/photon"

  config.vm.provision :shell, :path => "provisioning/install-base-deps.sh"
  config.vm.provision :shell, :path => "../provisioning/fix-vmware-tools.sh"
  config.vm.provision :shell, :path => "../provisioning/bootstrap-sysctl.sh"
  config.vm.provision :shell, :path => "../provisioning/bootstrap-environment.sh"
  config.vm.provision :shell, :inline => "echo 'export EDITOR=\"#{ENV['EDITOR'] || 'nano'}\"' >> /etc/environment"

  config.vm.provision :shell, :path => "../provisioning/bootstrap-rvm.sh", :args => "'#{ENV['RVM_PATCHSET']}'", :privileged => false

  vm_cpu = ENV['INTEGRATION_VM_CPU']
  vm_memory = ENV['INTEGRATION_VM_MEMORY']

  case ENV['INTEGRATION_VM_SIZE'].to_s.downcase
  when "large"
    vm_cpu ||= "4"
    vm_memory ||= "6144"
  when "medium"
    vm_cpu ||= "2"
    vm_memory ||= "4096"
  else
    vm_cpu ||= "2"
    vm_memory ||= "2048"
  end

  config.vm.provider :vmware_fusion do |v, override|
    v.vmx["memsize"] = vm_memory
    v.vmx["numvcpus"] = vm_cpu
  end

  config.cache.scope = :machine
  config.cache.auto_detect = false
  config.cache.enable :apt
  config.cache.enable :rvm
end
```

**Comment by eatstoomuchjam** [comment]
With the `vagrant-cachier` config removed, I still get the error message about networking.  I am running vagrant 1.7.2 and first tried this with vagrant-vmware-fusion 3.0.1 which I have since updated to 3.2.6 which didn't correct the problem.

**Comment by sethvargo** [comment]
Ping @frapposelli

**Comment by frapposelli** [comment]
Support for NFS is coming soon, the Vagrantfile embedded in the Photon box explicitly disables NFS because of that.

The `configure_networks` error is due to guest support in Vagrant, you can either compile Vagrant from master (support was merged yesterday https://github.com/mitchellh/vagrant/commit/2c247824e1af5f6474c8eee2ffbf9f0a53627be5) or install this plugin https://github.com/vmware/vagrant-guests-photon

**Comment by likeahoss** [comment]
Sooo in layman terms... how can I get NFS working in Photon?

**Comment by mitchellh** [comment]
Fixed in master.

@likeahoss we'll have a release next week.
